samples/net/https_client/src/main.c

```diff
@@ -168,7 +153,61 @@ int tls_setup(int fd)
 	return 0;
 }
 
-int main(void)
+static void on_net_event_l4_disconnected(void)
+{
+	printk("Disconnected from the network\n");
+}
+
+static void on_net_event_l4_connected(void)
+{
+	k_sem_give(&network_connected_sem);
+}
+
+static void l4_event_handler(struct net_mgmt_event_callback *cb,
+			     uint32_t event,
+			     struct net_if *iface)
+{
+	switch (event) {
+	case NET_EVENT_L4_CONNECTED:
+		printk("Network connectivity established and IP address assigned\n");
+		on_net_event_l4_connected();
+		break;
+	case NET_EVENT_L4_DISCONNECTED:
+		printk("Network connectivity lost, no IP address assigned\n");
+		on_net_event_l4_disconnected();
+		break;
+	default:
+		break;
+	}
+}
+
+static void ip_event_handler(struct net_mgmt_event_callback *cb,
+			     uint32_t event,
+			     struct net_if *iface)
+{
+	switch (event) {
+	case NET_EVENT_IPV4_ADDR_ADD:
```

Comment:
Since you already register for those events, maybe it'd make sense to actually print the assigned address? `cb->info` can be casted to `(const struct in_addr *)`/`(const struct in6_addr *)` (depending on the event) and then printed out (`net_addr_ntop()` can be used to convert address to a printable form).

Reply:
Actually this handler is not needed for the purpose of the sample, so it will be removed.